src/math/ray.zig

```diff
@@ -0,0 +1,218 @@
+const std = @import("std");
+
+const mach = @import("../main.zig");
+const testing = mach.testing;
+const math = mach.math;
+const vec = @import("vec.zig");
+
+fn maxDim(v: math.Vec3) u32 {
+    if (v.v[0] > v.v[1]) {
+        if (v.v[0] > v.v[2]) {
+            return 0;
+        } else {
+            return 2;
+        }
+    } else if (v.v[1] > v.v[2]) {
+        return 1;
+    } else {
+        return 0;
+    }
+}
+
+pub const RayHit = packed struct { u: f32, v: f32, w: f32, t: f32 };
+
+pub const Ray = struct {
```

Comment:
Can do. Followup question on that: currently my implementation also includes the fallback double precision as they handle it in the paper/reference. Which means it falls back to `f64` in case we encounter `(u == 0.0 or v == 0.0 or w == 0.0)` at first.

Do you want me to remove this for that change to keep the precision always fully conform, or keep it in anyway?

Reply:
It's pretty cool you have that fallback! I definitely like having that for the normal "I am using f32" case.

We could either:

1. Only employ it in the case of f32 being the underlying Vec element type
2. Scale it depending on the underlying Vec element type. e.g. if using `Ray3h` which is `f16`, then have an `f32` fallback.. if using `Ray3d` which is `f64` then have an `f128` fallback?
3. Always employ it irrespective of the underlying Vec element type. i.e. there's always an `f64` fallback.. even if the underlying Vec type is `f16` or `f64`

I could see arguments for either way.. I'm open to suggestions here, I guess I would reach for option 2?

Reply:
I like option 2 as well.